modules/gltf/src/gltf/create-gltf-model.ts

```diff
@@ -57,53 +58,61 @@ const fs = `
 #endif
 
   void main(void) {
-    fragmentColor = pbr_filterColor(vec4(0));
+    vec3 pos = pbr_vPosition;
+    fragmentColor = pbr_filterColor(vec4(1.0));
+    // fragmentColor = vec4(1.0, 0.0, 0.0, 1.0);
+    // fragmentColor = vec4(pos, 1.0);
   }
 `;
 
 export type CreateGLTFModelOptions = {
   id?: string;
-  topology?: PrimitiveTopology;
   vertexCount?: number;
-  attributes?: Record<string, any>;
+  geometry: Geometry;
   material: any;
   materialOptions: ParsePBRMaterialOptions;
   modelOptions?: Record<string, any>;
 };
 
 export function createGLTFModel(device: Device, options: CreateGLTFModelOptions): ModelNode {
-  const {id, attributes, material, topology, vertexCount, materialOptions, modelOptions} = options;
-
-  const parsedMaterial = parsePBRMaterial(device, material, attributes, materialOptions);
+  const {id, geometry, material, vertexCount, materialOptions, modelOptions} = options;
 
+  const parsedMaterial = parsePBRMaterial(device, material, geometry.attributes, materialOptions);
   log.info(4, 'createGLTFModel defines: ', parsedMaterial.defines)();
 
   // Calculate managedResources
   // TODO: Implement resource management logic that will
   // not deallocate resources/textures/buffers that are shared
   const managedResources = [];
   // managedResources.push(...parsedMaterial.generatedTextures);
-  managedResources.push(...Object.values(attributes).map((attribute) => attribute.buffer));
-
-  const model = new ModelNode({
-    managedResources,
-    model: new Model(device, {
-      id,
-      topology,
-      vertexCount,
-      modules: [pbr],
-      defines: parsedMaterial.defines,
-      parameters: parsedMaterial.parameters,
-      vs: addVersionToShader(device, vs),
-      fs: addVersionToShader(device, fs),
-      attributes,
-      bindings: parsedMaterial.bindings,
-      uniforms: parsedMaterial.uniforms,
-      ...modelOptions
-    })
-  });
-
-  return model;
+  // managedResources.push(...Object.values(attributes).map((attribute) => attribute.buffer));
+
+  const parameters: RenderPipelineParameters = {
+    depthWriteEnabled: true,
+    depthCompare: 'less',
+    depthFormat: 'depth24plus',
+    cullMode: 'back'
+  }
+
+  const modelProps = {
```

Comment:
Since we use string constant types etc it is my experience best to make sure these structures are typed when initializing them. That happens automatically if it is inline, but if declared separately it is best to import the type.